test/exception-handling.wast

```diff
@@ -102,6 +102,35 @@
       )
     )
 
+    ;; nested try-catch
+    (try
+      (do
+        (try
+          (do
+            (throw $e-i32 (i32.const 0))
+          )
+          (catch $e-i32
+            (drop (pop i32))
+          )
+          (catch_all)
+        )
+      )
+      (catch $e-i32
+        (drop (pop i32))
+      )
+      (catch_all
+        (try
+          (do
+            (throw $e-i32 (i32.const 0))
+          )
+          (catch $e-i32
+            (drop (pop i32))
+          )
+          (catch_all)
+        )
+      )
+    )
+
```

Comment:
This is not related to `delegate` but I just found out we don't have any tests that test nested try-catch, so I added one here